packages/react/src/component.tsx

```diff
@@ -6,13 +6,62 @@ import {
   generateMimeTypes,
 } from "uploadthing/client";
 import { useUploadThing } from "./useUploadThing";
-import type { FileRouter } from "uploadthing/server";
+import type { FileRouter, ExpandedRouteConfig } from "uploadthing/server";
 import type { DANGEROUS__uploadFiles } from "uploadthing/client";
 
 type EndpointHelper<TRouter extends void | FileRouter> = void extends TRouter
   ? "YOU FORGOT TO PASS THE GENERIC"
   : keyof TRouter;
 
+const generatePermittedFileTypes = (config?: ExpandedRouteConfig) => {
+  const fileTypes = config ? Object.keys(config) : [];
+
+  const maxFileCount = config
+    ? Object.values(config).map((v) => v.maxFileCount)
+    : [];
+
+  return { fileTypes, multiple: maxFileCount.some((v) => v && v > 1) };
+};
+
+const capitalizeStart = (str: string) => {
+  return str.charAt(0).toUpperCase() + str.slice(1);
+};
+
+const INTERNAL_doFormatting = (config?: ExpandedRouteConfig): string => {
+  if (!config) return "";
+
+  const allowedTypes = Object.keys(config) as (keyof ExpandedRouteConfig)[];
+
+  const formattedTypes = allowedTypes.map((f) =>
+    f === "blob" ? "files" : `${f}s`
+  );
+
+  // Format multi-type uploader label as "Supports videos, images and files";
+  if (formattedTypes.length > 1) {
+    const lastType = formattedTypes.pop();
+    return `${formattedTypes.join(", ")} and ${lastType}`;
+  }
+
+  // Single type uploader label
+  const key = allowedTypes[0];
+  const formattedKey = formattedTypes[0];
+
+  // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+  const { maxFileSize, maxFileCount } = config[key]!;
+
+  if (maxFileCount && maxFileCount > 1) {
+    return `${formattedKey} up to ${maxFileSize}, max ${maxFileCount}`;
+  } else {
+    return `${formattedKey.slice(0, -1)} (${maxFileSize})`;
```

Comment:
Not a huge fan of pluralizing by default, and slicing off the s if needed, seems kind of backwards?